www/js/control/ProfileSettings.jsx

```diff
@@ -65,15 +49,18 @@ const ProfileSettings = () => {
     const [forceStateVis, setForceStateVis] = useState(false);
     const [permitVis, setPermitVis] = useState(false);
     const [logoutVis, setLogoutVis] = useState(false);
-    const [dataPendingVis, setDataPendingVis] = useState(false);
-    const [dataPushedVis, setDataPushedVis] = useState(false);
     const [invalidateSuccessVis, setInvalidateSuccessVis] = useState(false);
     const [noConsentVis, setNoConsentVis] = useState(false);
     const [noConsentMessageVis, setNoConsentMessageVis] = useState(false);
     const [consentVis, setConsentVis] = useState(false);
     const [dateDumpVis, setDateDumpVis] = useState(false);
     const [privacyVis, setPrivacyVis] = useState(false);
+    const [showingSensed, setShowingSensed] = useState(false);
+    const [showingLog, setShowingLog] = useState(false);
+    const [editSync, setEditSync] = useState(false);
+    const [editCollection, setEditCollection] = useState(false);
```

Comment:
I'm a bit confused about the interaction between `editCollection` and `editCollectionConfig`.
I see that there is an editCollectionConfig which sets `editCollection` to true, which presumably opens the popup for the edit. But then what does `editCollection` do, and how does it interact with the `refreshCollectionSettings` in the `useEffect` below, and in the `control.update.complete event broadcast` from the helper

Reply:
The `control.update.complete` event broadcast cannot be relied on anymore, as it was intertwined with Angular routing. 

`editCollectionConfig` is what is called when the user clicks the pencil, setting `editCollection` to true and showing the editor menu. When the menu is closed, we want `refreshCollectionSettings` to be called so the `useEffect` calls `refreshCollectionSettings` if `editCollection` changes (it will become false when the editor is closed). This way, the UI for the settings is updates without having to rely on the broadcast events from angular. 

Does that help explain things?